doc/neps/nep-0052-python-api-cleanup.rst

```diff
@@ -122,53 +147,77 @@ probably. Here is a representative set of examples:
   and other issues for ``maximum_sctype`` and related functions).
 - Business day functionality can likely be removed (unclear if it needs
   splitting out like was done for ``np.financial``).
-- The ``np.compat`` namespace can be removed.
-- There are lots of one-off functions that can be deprecated and removed, such as
-  ``real_if_close`` (see `gh-11375 <https://github.com/numpy/numpy/issues/11375>`__).
-  These can be identified via triaging the issue tracker and/or via going
-  through the main namespace manually.
 
-There are new namespaces for warnings/exceptions (``np.exceptions``) and for dtype-related
-functionality (``np.types``). NumPy 2.0 is a good opportunity to move things
-there from the main namespace.
+S: I think `np.financial` maintainer was interested in adopting?
+
+- The ``np.compat`` namespace, used during the Python 2 to 3 transition, will be removed.
+- Functions that are narrow in scope, with very few public use-cases,
+  will be removed.  See, e.g.
+  ``real_if_close`` (`gh-11375 <https://github.com/numpy/numpy/issues/11375>`__).
+  These will have to be identified manually and by issue triage.
+
+New namespaces are introduced for warnings/exceptions (``np.exceptions``) and for dtype-related
+functionality (``np.types``). NumPy 2.0 is a good opportunity to
+popular these submodules from the main namespace.
 
-Functionality that gets a fair amount of usage but has a preferred alternative
+.. S: Has the ``np.types`` name been fixed? Wonder if we're going to
+   create confusion with that name.
+
+Functionality that is widely used but has a preferred alternative
 may be hidden by not including it in ``__dir__``, rather than deprecating it. A
 ``.. legacy::`` directory may be used to mark such functionality in the
 documentation.
 
-A more comprehensive test will be added to the test suite to ensure that we won't get
-any more accidental additions to any namespace - every new entry will need to be
-allow-listed.
+.. S: Perhaps we should give ourselves more freedom here: delete
+   unused functions, deprecate common functions for which an
+   alternative is recommended. We can raise a very helpful exception
+   in these cases, telling the developer exactly what to change.
```

Comment:
Good point. Ideally we get as many deprecations as possible into 1.25.0, but for really random stuff we could just remove it straight away.